internal/indexer/block_context.go

```diff
@@ -25,6 +27,9 @@ func MakeBlockContext(s *storage.Storage, client *chain.WsClient) *blockContext
 	bc.batch = s.NewBatch()
 	bc.client = client
 	bc.addressStats = make(map[string]*storage.AddressStats)
+	tn := time.Unix(int64(bc.age), 0)
+	year, week := tn.ISOWeek()
+	bc.WeekStats = s.GetWeekStats(year, week)
```

Comment:
I think we can move this to some new function that would be called from process function and would be run in a goroutine
```
func UpdateValidatorStats() {
	tn := time.Unix(int64(bc.age), 0)
	year, week := tn.ISOWeek()
	weekStats := s.GetWeekStats(year, week)
    weekStats.AddPbftBlock(block)
	bc.batch.UpdateWeekStats(weekStats)
}
```